cli_tools/common/verify/verify.go

```diff
@@ -0,0 +1,36 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package verify
+
+import "fmt"
+
+// GreaterThanOrEqualTo verifies that value is greater than or equal to limit,
+// and panics if the verification fails.
+func GreaterThanOrEqualTo(value int, limit int) {
+	if value < limit {
+		panic(fmt.Sprintf("Expected %d >= %d", value, limit))
```

Comment:
panic is supposed to be used for runtime errors that a program can't recover from, errors it doesn't know how to handle or API errors (made by programmers): https://gobyexample.com/panic

> Mostly we use it to fail fast on errors that shouldn’t occur during normal operation, or that we aren’t prepared to handle gracefully.

I think for generic validation it's better to explicitely return an error. These functions might be used for validating user data in the future where panic is not appropriate. Right now, the use of it in new*Release() makes sense as it's guarding against programmer errors, but having it in a generic verify package opens up these funcs to be used inappropriately.

Reply:
Yeah, like you mentioned, the `verify` package is intended to verify preconditions and assumptions, allowing  *programming* errors to be surfaced. It's not intended to be used with unsanitized user input. 

`These functions might be used for validating user data in the future`

Gosh, I hope not! My intention was to create a small, well-defined public API that allows the internal details (such as struct members) to evolve over time. If you have some specific usecases that are not covered by this public API, I'd be happy to brainstorm what they'd look like.

Reply:
If this package will only hold code to protect against programming errors, then `assert` would be a better name than `verify`. `verify` can apply to both user and programming errors, while `assert` is clear it's about programming errors.

Reply:
sgtm (done)